src/Authentication/Traits/HasAccessTokens.php

```diff
@@ -165,4 +171,46 @@ public function setAccessToken(?AccessToken $accessToken): self
 
         return $this;
     }
+
+    /**
+     * Checks if the provided Access Token has expired.
+     *
+     * @return bool Returns true if Access Token has expired, false if not
+     */
+    public function hasAccessTokenExpired(?AccessToken $accessToken): bool
```

Comment:
```suggestion
    public function hasAccessTokenExpired(AccessToken $accessToken): bool
```
I don't think we should allow null.

Reply:
Yeah, that make sense.

I also noticed that  `setAccessToken()` and `setHmacToken()` are allowing a null value, and they probably shouldn't unless there is a reason to clear the access/hmac token.

Reply:
Well, the `$accessToken` in `setAccessToken()` and `setHmacToken()` can be `null`, because the user may simply have no token (e.g. see `AccessTokens::loginById()`). 

But this new method doesn't rely on the `$this->currentAccessToken` property, but is rather intended to be used to administer tokens when we use `accessTokens()`. So, my thinking was the `$accessToken` should not be optional.